x-pack/plugins/security_solution/public/timelines/components/edit_data_provider/components/controlled_default_input.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useMemo, useEffect, useCallback } from 'react';
+
+import { EuiFieldText } from '@elastic/eui';
+
+import type { DataProviderType } from '../../timeline/data_providers/data_provider';
+import { isStringOrNumberArray } from '../../timeline/helpers';
+import { isValueFieldInvalid, sanatizeValue } from '../helpers';
+import * as i18n from '../translations';
+
+interface ControlledDataProviderInput {
+  disableButtonCallback: (disableButton: boolean) => void;
+  onChangeCallback: (value: string | number | string[]) => void;
+  type: DataProviderType;
+  value: string | number | Array<string | number>;
+}
+
+const VALUE_INPUT_CLASS_NAME = 'edit-data-provider-value';
+
+export const ControlledDefaultInput = ({
+  value,
+  type,
+  disableButtonCallback,
+  onChangeCallback,
+}: ControlledDataProviderInput) => {
+  const [primitiveValue, setPrimitiveValue] = useState<string | number>(getDefaultValue(value));
+
+  const isInvalid = useMemo(
+    () => isValueFieldInvalid(type, primitiveValue),
+    [type, primitiveValue]
+  );
+
+  useEffect(() => {
+    onChangeCallback(primitiveValue);
+    disableButtonCallback(isInvalid);
+  }, [primitiveValue, isInvalid, onChangeCallback, disableButtonCallback]);
+
+  const onValueChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setPrimitiveValue(e.target.value);
+  }, []);
+
+  return (
+    <EuiFieldText
+      className={VALUE_INPUT_CLASS_NAME}
+      onChange={onValueChange}
+      placeholder={i18n.VALUE}
+      value={sanatizeValue(primitiveValue)}
```

Comment:
In the `useEffect` above, `onChangeCallback` is invoked with the raw `primitiveValue`:

```ts
  useEffect(() => {
    onChangeCallback(primitiveValue);
    // ...
```

Should the `onChangeCallback`receive the raw `primitiveValue`, or the `sanatizeValue(primitiveValue)` version of the value, as provided here to the `EuiFieldText`?

Reply:
Youre right! Ive updated it to take the sanitized value